pyscf/tools/finite_diff.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+# Copyright 2025 The PySCF Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+Finite difference driver
+'''
+
+import numpy as np
+from pyscf import gto
+from pyscf import lib
+from pyscf.lib import logger
+from pyscf.grad.rhf import GradientsBase
+from pyscf.hessian.rhf import HessianBase
+
+def kernel(method, displacement=1e-2):
```

Comment:
1e-2 seems big

Reply:
I disagree. Also, (assuming same units), this is the same default as in [OpenMolcas](https://molcas.gitlab.io/OpenMolcas/sphinx/users.guide/programs/alaska.html). In Orca it appears to be 0.005 Bohr.

Reply:
1e-2 typically gives an error ~1e-4. A smaller displacement would require tightening the convergence tolerance of the underlying methods, at least to 1e-6. Otherwise, the errors in the underlying methods would dominate the finite-difference error. 1e-6 is already near the limit of many post-HF methods. This is because the orbital coefficients usually have errors on the order of the square root of the HF convergence threshold.

When computing the Hessian via finite differences, the gradients tend to carry the error in the orbitals, or even larger. ~1e-2 is almost the smallest you can use to get reliable Hessian.